src/store/index.js

```diff
@@ -179,41 +173,55 @@ export default createStore({
               : setTimeout(() => commit('enableIframeWallet'), 15000);
             commit('useSdkWallet');
 
-            const scannerConnection = await BrowserWindowMessageConnection({
-              connectionInfo: { id: 'spy' },
-            });
-            const detector = await WalletDetector({ connection: scannerConnection });
+            let resolve = null;
+            let rejected = (e) => {
+              throw e;
+            };
+            let stopScan = null;
 
-            await new Promise((resolve, rejected) => {
-              detector.scan(async ({ wallets }) => {
-                const detectedWallet = Object.values(wallets).find((w) => w.name === wallet.name);
-                if (!detectedWallet) return;
-                clearInterval(webWalletTimeout);
-
-                try {
-                  await sdk.connectToWallet(await detectedWallet.getConnection());
-                  await sdk.subscribeAddress('subscribe', 'current');
-                } catch (e) {
-                  if (e.message !== 'Operation rejected by user') {
-                    dispatch('showUnknownError', e);
-                    dispatch('disconnectWallet');
-                  }
-                  rejected(e);
-                  return;
-                }
-
-                const currentAccountAddress = sdk.rpcClient.getCurrentAccount();
+            const connectWallet = async (wallet) => {
+              try {
+                const { networkId } = await sdk.connectToWallet(wallet.getConnection());
+                const ret = await sdk.subscribeAddress('subscribe', 'connected');
+                const { address: { current } } = ret;
+                const currentAccountAddress = Object.keys(current)[0];
                 if (!currentAccountAddress) return;
-                detector.stopScan();
-                const { networkId: walletNetworkId } = sdk.rpcClient.info;
+                stopScan?.();
                 commit('setAddress', currentAccountAddress);
-                dispatch('selectNetwork', walletNetworkId);
-                resolve(currentAccountAddress);
+                dispatch('selectNetwork', networkId);
+                resolve?.(currentAccountAddress);
+              } catch (e) {
+                if (e.message !== 'Operation rejected by user') {
+                  dispatch('showUnknownError', e);
+                  dispatch('disconnectWallet');
+                }
+                rejected(e);
+              }
+            };
+            if (walletObj.getConnection) {
+              await connectWallet(walletObj);
+            } else {
+              const handleWallet = async ({ wallets }) => {
+                const detectedWalletObject = Object.values(wallets).find(
+                  (wallet) => wallet.info.name === walletObj.info.name,
+                );
+                if (!detectedWalletObject) return;
+                clearInterval(webWalletTimeout);
+                // TODO: this will ask for permission every time
```

Comment:
still open todo?

Reply:
if it is still the issue, probably it is on SW side